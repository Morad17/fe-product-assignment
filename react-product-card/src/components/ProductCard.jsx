import React, { useState } from "react";
import ProductImage from "../assets/images/image-product-desktop.jpg";
import MobileProductImage from "../assets/images/image-product-mobile.jpg";
import cartIcon from "../assets/images/icon-cart.svg";

const ProductCard = () => {

    const [addedToCart, setAddedToCart] = useState(false);

  return (
    <div className="product-card">
      <div className="product-card-image">
        <picture>
          <source srcSet={MobileProductImage} media="(max-width: 425px)" />
          <img src={ProductImage} alt="Product" />
        </picture>
      </div>
      <div className="product-card-info">
        <div className="product-card-wrapper">
          <h3 className="product-card-category">PERFUME</h3>
          <div className="product-card-name-main">
            <h1 className="product-card-name">
              Gabrielle Esscence Eau De Parfum
            </h1>
          </div>

          <p className="product-card-description">
            A floral, solar and voluptuous interpretation composed by Olivier
            Polge, Perfume-Creater for the House of CHANEL.
          </p>
          <div className="product-card-price">
            <span className="product-card-price-current">$149.99</span>
            <span className="product-card-price-original">
              <s>$169.99</s>
            </span>
          </div>
          {
            addedToCart ? 
          <button id="addToCart"onClick={()=> setAddedToCart(!addedToCart)}className="product-card-btn active">
            Added
          </button> :
           <button id="addToCart" onClick={()=> setAddedToCart(!addedToCart)}className="product-card-btn">
            <img className="svg" src={cartIcon} alt="" /> Add to Cart
          </button>
          }
        </div>
      </div>
    </div>
  );
};

export default ProductCard;
