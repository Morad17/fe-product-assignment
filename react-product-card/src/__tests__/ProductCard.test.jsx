import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ProductCard from "../components/ProductCard";

test('shows "Added" button after clicking Add to Cart three times', () => {
  const { getByRole, asFragment, queryByText } = render(<ProductCard />);
  
  // Find Add To Cart Button
  let button = getByRole("button", { name: /add to cart/i });

  // Simulates Clicking three times
  fireEvent.click(button);
  button = getByRole("button");
  fireEvent.click(button);
  button = getByRole("button");
  fireEvent.click(button);

  // Check to see if button text has Added
  expect(queryByText(/added/i)).toBeInTheDocument();

  // Snapshot result
  expect(asFragment()).toMatchSnapshot();
});