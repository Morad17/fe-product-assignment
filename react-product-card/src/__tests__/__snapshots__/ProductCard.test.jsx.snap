// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows "Added" button after clicking Add to Cart three times 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card-image"
    >
      <picture>
        <source
          media="(max-width: 425px)"
          srcset="image-product-mobile.jpg"
        />
        <img
          alt="Product"
          src="image-product-desktop.jpg"
        />
      </picture>
    </div>
    <div
      class="product-card-info"
    >
      <div
        class="product-card-wrapper"
      >
        <h3
          class="product-card-category"
        >
          PERFUME
        </h3>
        <div
          class="product-card-name-main"
        >
          <h1
            class="product-card-name"
          >
            Gabrielle Esscence Eau De Parfum
          </h1>
        </div>
        <p
          class="product-card-description"
        >
          A floral, solar and voluptuous interpretation composed by Olivier Polge, Perfume-Creater for the House of CHANEL.
        </p>
        <div
          class="product-card-price"
        >
          <span
            class="product-card-price-current"
          >
            $149.99
          </span>
          <span
            class="product-card-price-original"
          >
            <s>
              $169.99
            </s>
          </span>
        </div>
        <button
          class="product-card-btn active"
          id="addToCart"
        >
          Added
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
